docs/en/02-consensus.md

```diff
@@ -2,37 +2,45 @@
+The selected proposer proposes a block. The unconfirmed transactions that have not yet been added in the blockchain are shared via P2P between the nodes nodes in the network and stored in an area of each node called the mempool. The node selected as the proposer generates a block from the unconfirmed transactions remaining in its mempool and proposes it to the validators.
 
-## VRF-based Consensus Group Election
+### Block verification
 
-VRF is an algorithm for generating a hash value $t$ that can be used as a cryptographic pseudo-random number. The differences between VRF and typical hash functions or pseudo-random number generators are that only the owner of the private key can generate the hash value $t$, and anyone with the corresponding public key can verify the correctness of the hash value.
+The validators validate the block proposed by the proposer. Each validator votes on whether the block is correct, and Tendermint-BFT replicates the votes to the other validators. If more than $\frac{2}{3}$ of all validators vote in favor of the block, the block is officially approved. On the other hand, if a quorum is not reached, the proposed block is rejected, and a new round of elections or voting is started over.
 
-A VRF hash generator $k$ generates a proof $\pi$ (VRF Proof) from the message $m$ using its private key $S_k$ as in Equation `1.`. Here, the hash value $t$ can be acquired from the proof $pi$ using Equation. `2.` On the other hand, to verify that the hash value $t$ was generated by the owner of the private key $S_k$ based on the message $m$, the verifier applies the public key $P_k$ for $S_k$, $m$, and $\pi$ to Equation `3.` to verify that both hash values are identical.
+> TIP: In Tendermint-BFT, this re-election process can be routed to a particular stage of the election process, depending on reasons for rejection.
+
+![VRF-based block generation round](../static/consensus/vrf_based_round.png)
+
+## VRF-based proposer election
+
+VRF is an algorithm for generating a hash value $t$ that can be used as a cryptographic pseudorandom number. The difference between VRF and typical hash functions or pseudorandom number generators is that only the owner of the private key can generate the hash value $t$, and anyone with the corresponding public key can verify the correctness of the hash value.
+
+A VRF hash generator $k$ generates a proof $\pi$ (VRF Proof) from the message $m$ using its private key $S_k$ as in equation `1.`. Here the hash value $t$ can be obtained from the proof $pi$ using equation. `2.` On the other hand, to verify that the hash value $t$ was generated by the owner of the private key $S_k$ based on the message $m$, the verifier applies the public key $P_k$ for $S_k$, $m$, and $\pi$ to equation `3.` to verify that both hash values are identical.
 
 1. $\pi = {\rm vrf\\_prove}(S_k, m)$
 2. $t = {\rm vrf\\_proof\\_to\\_hash}(\pi)$
 3. ${\rm vrf\\_proof\\_to\\_hash}(\pi) \overset{\text{?}}{=} {\rm vrf\\_verify}(P_k, m, \pi)$
```

Comment:
How about writing something like:

```math
\begin{align*}
   \pi & = & {\rm vrf\\_prove}(S_k, m) \tag{1}\\
   t & = & {\rm vrf\\_proof\\_to\\_hash}(\pi) \tag{2}\\
   {\rm vrf\\_proof\\_to\\_hash}(\pi) & \overset{\text{?}} {=} & {\rm vrf\\_verify}(P_k, m, \pi) \tag{3}
\end{align*}
```

Reply:
I see that the auto-numbering doesn't work on GitHub (katex-based?), but `tag`s can be used. Thanks.